src/pfe/portal/modules/LoadRunner.js

```diff
@@ -341,48 +341,42 @@ module.exports = class LoadRunner {
-
+    const healthCentercommand = `export javapid=(\`pidof java\`); java -jar $JAVA_HOME/jre/lib/ext/healthcenter.jar ID=\${javapid[-1]} level=headless -Dcom.ibm.java.diagnostics.healthcenter.headless.run.number.of.runs=1 -Dcom.ibm.java.diagnostics.healthcenter.headless.run.duration=${durationInMins} -Dcom.ibm.java.diagnostics.healthcenter.headless.output.directory=${loadTestDir} -Dcom.ibm.diagnostics.healthcenter.data.memory=off -Dcom.ibm.diagnostics.healthcenter.data.memorycounters=off -Dcom.ibm.diagnostics.healthcenter.data.cpu=off -Dcom.ibm.diagnostics.healthcenter.data.environment=off -Dcom.ibm.diagnostics.healthcenter.data.locking=off -Dcom.ibm.diagnostics.healthcenter.data.memory=off -Dcom.ibm.diagnostics.healthcenter.data.threads=off`;
+    const healthCenterArray = ['bash', '-c', healthCentercommand];
     try {
-      await docker.exec(this.project, ['bash', '-c', command]); 
+      await cwUtils.spawnContainerProcess(this.project, healthCenterArray);
     } catch (error) {
       log.error(error)
     }
 
-    const projectID = this.project.projectID;
-
-    let options = {
-      host: "localhost",
-      port: 9090,
-      path: `/api/v1/projects/${projectID}/profiling/${this.metricsFolder}`,
-      method: 'GET',
-    };
-    
-    setTimeout(() => this.getJavaHealthCenterData(options, projectID, this.metricsFolder, 1), duration * 1000);
+    setTimeout(() => this.getJavaHealthCenterData(1), duration * 1000);
   }
 
-  async getJavaHealthCenterData(options, projectID, timestamp, counter) {
+  async getJavaHealthCenterData(counter) {
     if (counter > 20) {
       log.error("Failed to save .hcd file");
       return;
     }
-    const apiOptions = options;
-    const res = await cwUtils.asyncHttpRequest(apiOptions);
-    if (res.statusCode !== 200) {
+    try {
+      await this.project.getProfilingByTime(this.metricsFolder);
+    } catch (error) {
+      log.error(error);
       log.info(`.hcd file not found, trying again. Attempt ${counter}/20`);
-      setTimeout(() => this.getJavaHealthCenterData(apiOptions, projectID, timestamp, counter + 1), 3000);
-    } else {
-      log.info(".hcd copied to PFE");
-      const data = { projectID: projectID,  status: 'hcdReady', timestamp: timestamp};
-      this.user.uiSocket.emit('runloadStatusChanged', data);
+      setTimeout(() => this.getJavaHealthCenterData(counter + 1), 3000);
```

Comment:
Could you add a null check for project here? Then we won't keep retrying 20 times if someone cancels the run and we null out the this.project object.

Reply:
Change is in